store/ssmstore.go

```diff
@@ -284,19 +340,47 @@ func (s *SSMStore) Delete(ctx context.Context, id SecretId) error {
 }
 
 func (s *SSMStore) DeleteTags(ctx context.Context, id SecretId, tagKeys []string) error {
+	err := s.checkIfDeletingRequiredTags(ctx, tagKeys)
+	if err != nil {
+		return err
+	}
+
 	removeTagsInput := &ssm.RemoveTagsFromResourceInput{
 		ResourceType: types.ResourceTypeForTaggingParameter,
 		ResourceId:   aws.String(s.idToName(id)),
 		TagKeys:      tagKeys,
 	}
-	_, err := s.svc.RemoveTagsFromResource(ctx, removeTagsInput)
+	_, err = s.svc.RemoveTagsFromResource(ctx, removeTagsInput)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
 
+func (s *SSMStore) checkIfDeletingRequiredTags(ctx context.Context, tagKeys []string) error {
+	requiredTags, err := requiredTags(ctx, s)
+	if err != nil {
+		return err
+	}
+	tags := make(map[string]any)
+	for _, key := range tagKeys {
+		tags[key] = struct{}{}
+	}
```

Comment:
❓ I see that we are creating tags as a map/set in this case. Is it because the tagKeys could be duplicated and thus, we are creating a map/set in this case? Curious if it is possible that it could be duplicated

Reply:
The user running `chamber tag delete` could list a tag more than once. But, the reason for the set here is just to make it easier to check if a required tag was provided - instead of having to iterate through the `tagKeys` slice for every required tag, it's just a set lookup.

```
	var foundTags []string
	for _, requiredTag := range requiredTags {
		if _, ok := tags[requiredTag]; ok {
			foundTags = append(foundTags, requiredTag)
		}
	}
```

The code could instead make a set for `requiredKeys` and then iterate through the original `tagKeys` slice (is this one required? is this next one required?) ... although then it could report a required key more than once, if the user listed it more than once to delete.

Just to check, I ran `chamber tag delete` on a secret and listed a tag twice, and it was removed without error.